### DISCUSSION TITLE
Target audience of the library

### DISCUSSION THREAD

**Discussion by ecyrbe** [decs/typeschema#7] [Ideas]
hello @decs ,

tell me if i'm wrong, but i expect typeschema target audience to be library implementor like :
- trpc
- hono
- zodios
- ts-rest
- hook-form

i don't expect application level to need this, because being tied to only one validation library is fine for an application.

if that's the case, i think it would be a good idea to let the  library authors to pick what adapters they want to include by default.
because having a huge list of resolvers in the typescript registry can deeply impact typescript performances.
Allowing library author to cherry pick the minimum set of libs would allow to have correct performances if they support only 2 or 3 libraries by default, while still allowing their users to add the one they want if it's not in the library default list.

So i propose to split the library in a monorepo, with all adapters in their respective packages.

what do you think ?

**Comment by astahmer** [comment]
This also leads to [issues](https://github.com/partykit/partykit/issues/254) where you can't use typeschema in a bundler without listing all of the externals
https://github.com/astahmer/partykit-typeschema/blob/main/partykit.json#L10-L16

just like the special vite integration that shouldnt be needed imo
https://github.com/decs/typeschema/blob/main/src/vite/index.ts#L6-L13

**Comment by decs** [comment]
Solved as part of `v0.13.0`
